Tests/HarmonizeTests/Tests/LintRulesExamplesTests.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  LintRulesExamplesTests.swift
+//  Harmonize
+//
+//  Copyright (c) Perry Street Software 2024. All Rights Reserved.
+//
+
+import Foundation
+import Harmonize
+import XCTest
+
+final class LintRulesExamplesTests: XCTestCase {
+    override var testRunClass: AnyClass? {
+        return AssertionsFailuresTests.ExpectedFailureTestCaseRun.self
+    }
+    
+    func testNoSideEffects() throws {
+        Harmonize.on { Self.ViewModel }
+            .classes()
+            .initializers()
+            .withFunctionCalls { functionCalls in
+                let closures = functionCalls.compactMap(\.closure)
+                let hasAssignmentToSelf = closures.filter {
+                    $0.body?.assignments.contains { $0.target.contains("self") } == true
+                }
+                
+                return hasAssignmentToSelf.isNotEmpty
```

Comment:
I'm not entirely sure about that, but didn't we agree that assignments are allowed in initializers but side-effects are not?

Reply:
This was just an example! Yes, assignments are fine **but** assignments within a closure might be a side-effect -- this is where this example lint rule is acting.